torch/distributed/_tensor/random.py

```diff
@@ -301,3 +261,158 @@ def is_rng_supported_mesh(device_mesh: DeviceMesh) -> bool:
             f"DTensor random operators may not have complete support on {device_mesh.device_type} device mesh"
         )
         return False
+
+
+def manual_seed(seed: int, device_mesh: DeviceMesh, tp_dim: int=0) -> None:
+    """Sets the seed for generating random numbers for the calling rank.
+
+    Args:
+        seed (int): The desired seed.
+        device_mesh (:class:`DeviceMesh`): The device mesh to set the seed.
+        tp_dim (int, optional): The mesh dimension where to apply Tensor Parallel
+            Default: 0
+
+    Returns:
+        None
+
+    .. warning::
+        When calling this function, :func:`manual_seed` must be called from all ranks of the
+        default `ProcessGroup` even if some ranks may not be a part of the `device_mesh`,
+        with the same `seed` value.
+        If ``device_mesh`` is a sub-mesh and the calling rank is not a part of it,
+        `manual_seed` will not set its GPU device's generator seed.
+        Current implementation only supports a GPU device mesh.
+    """
+    # allgather the seed from rank 0 over the default PG
+    object_list = [seed] * dist.get_world_size()
+    dist.all_gather_object(object_list, seed)
+    for rank, object in enumerate(object_list):
+        if seed != int(object):
+            raise RuntimeError(
+                f"calling manual_seed function over {device_mesh} but received different seed values on ranks:",
+                f"seed on rank {dist.get_rank()} is {seed}, and seed on rank {rank} is {object}!",
+            )
+
+    # the current rank is in mesh
+    if device_mesh.get_coordinate() is not None:
+        if isinstance(_parallel_rng_style, MegatronRNG):
+            _parallel_rng_style._manual_seed(device_mesh, seed, tp_dim)
+        elif isinstance(_parallel_rng_style, DTensorRNG):
+            _parallel_rng_style._manual_seed(seed)
+        else:
+            raise RuntimeError("Unknown parallel RNG style!")
+
+
+# TODO: change this to ABC
+class ParallelRNGStyle:
+    def __init__(self):
+        pass
```

Comment:
Do I understand correctly that you want to do something like this:
```
# non-TP style for initialization
_parallel_rng_style = DTensorParallelRNG()
dtensor.uniform_()

# use TP style for random dropout
_parallel_rng_style = TensorParallelRNG()
dtensor.dropout()
```

We offer this flexibility by assigning different values to global `_parallel_rng_style`.

Reply:
Do we want to explicitly expose this to users? I thought this will just be one entry of `DeviceMesh`?

Reply:
> Do we want to explicitly expose this to users? I thought this will just be one entry of `DeviceMesh`?

No this shouldn't be facing to users directly but we can have this within TP's API implementation. I'm still thinking about where to put this variable. Currently it is a module-level variable within `random.py` but maybe we can put it inside device mesh @wanchaol WDYT?

Reply:
updated the code snippet. Getting rid of the `_parallel_rng_style` variable seems to be simpler.

Reply:
I think it should just be the module level global variable. Inside TP's API it can override this to take TPRNG.

Also, I think we should NOT make this a DeviceMesh attribute as it's already a global attribute, we should make sure RNGTrackers are lazily initialized so that we could put those initialization logic in different APIs (i.e. `distribute_tensor` or `parallelize_module`)